test/framework/finalizers_helpers.go

```diff
@@ -122,10 +124,11 @@ func getObjectsWithFinalizers(ctx context.Context, proxy ClusterProxy, namespace
 		Expect(err).ToNot(HaveOccurred())
 
 		setFinalizers := obj.GetFinalizers()
+		expectedFinalizers := allFinalizerAssertions[node.Object.Kind](types.NamespacedName{Namespace: node.Object.Namespace, Name: node.Object.Name})
```

Comment:
There seems to be a nil pointer here

Reply:
```
In [It] at: /usr/local/go/src/runtime/panic.go:261 @ 05/28/24 15:03:26.623

runtime error: invalid memory address or nil pointer dereference

Full Stack Trace
  sigs.k8s.io/cluster-api/test/framework.getObjectsWithFinalizers({0x2b31788, 0xc0007afe50}, {0x2b43d30, 0xc000762360}, {0xc0008368a0, 0x12}, 0xc001681110, 0xc001702768)
  	/home/prow/go/src/sigs.k8s.io/cluster-api/test/framework/finalizers_helpers.go:127 +0x4cb
  sigs.k8s.io/cluster-api/test/framework.ValidateFinalizersResilience({0x2b31788, 0xc0007afe50}, {0x2b43d30, 0xc000762360}, {0xc0008368a0, 0x12}, {0xc000836918, 0x12}, 0xc001702768, {0xc0014151f0, ...})
  	/home/prow/go/src/sigs.k8s.io/cluster-api/test/framework/finalizers_helpers.go:75 +0xee
  sigs.k8s.io/cluster-api/test/e2e.init.func30.1.1({0x2b43d30, 0xc000762360}, {0xc0008368a0, 0x12}, {0xc000836918, 0x12})
  	/home/prow/go/src/sigs.k8s.io/cluster-api/test/e2e/quick_start_test.go:105 +0x353
  sigs.k8s.io/cluster-api/test/e2e.QuickStartSpec.func2.1()
  	/home/prow/go/src/sigs.k8s.io/cluster-api/test/e2e/quick_start.go:153 +0x46
  sigs.k8s.io/cluster-api/test/framework/clusterctl.ApplyCustomClusterTemplateAndWait({_, _}, {{0x2b43d30, 0xc000762360}, {0xc0009d0001, 0x74ef, 0x74f0}, {0xc000836918, 0x12}, {0xc0008368a0, ...}, ...}, ...)
  	/home/prow/go/src/sigs.k8s.io/cluster-api/test/framework/clusterctl/clusterctl_helpers.go:472 +0x163a
  sigs.k8s.io/cluster-api/test/framework/clusterctl.ApplyClusterTemplateAndWait({_, _}, {{0x2b43d30, 0xc000762360}, {{0xc000848bd0, 0x22}, {0xc00015a916, 0x31}, {0xc00015a948, 0x17}, ...}, ...}, ...)
  	/home/prow/go/src/sigs.k8s.io/cluster-api/test/framework/clusterctl/clusterctl_helpers.go:350 +0x9f1
  sigs.k8s.io/cluster-api/test/e2e.QuickStartSpec.func2()
  	/home/prow/go/src/sigs.k8s.io/cluster-api/test/e2e/quick_start.go:133 +0x748

```

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/10693/pull-cluster-api-e2e-blocking-main/1795467930796822528

Reply:
Should be fixed now, but I have a follow up to investigate